docs/DeveloperGuide.md

```diff
@@ -1,29 +1,342 @@
+
+## 3. Design & implementation
+
+### 3.1. Architecture
+
+<p align="center">
+    <img src="team/images/Architecture.png">
+    <i>Figure 1: Architecture Diagram for MyLedger</i>
+</p>
+
+The Architecture Diagram shown above is a high-level components within MyLedger. The ```MyLedger``` class contains the main method which
+is responsible for:
+
+1. When MyLedger is launched, it will initialize the ```Storage``` to load the saved expenditures from the textfile and  ```Ui``` to print
+   the welcome message.
+2. When MyLedger is executing, it receives input for the user and sends it to ```Storage``` then ```Command``` which carries out the various
+   commands. 
+3. After the command has been carried out ```Command``` sends the result back to ```MyLedger.main()``` which would print the message back to the user.
+
+The other components of MyLedger include:
+
+* ```Ui```: The user interface that prints the welcome and help message.
+* ```Parser```: Parser which process the user's command and calls the specific command.
+* ```Command```: Executes the command given.
+* ```Expenditure```: Constructs an expenditure and is added to ```ExpenditureList```.
+* ```ExpenditureList```: An ArrayList of the current expenditures.
+* ```Storage```: Uses ```MyLedger_inputs.txt``` to initialize ```ExpenditureList```, updates ```MyLedger_inputs.txt``` whenever ```ExpenditureList```
+  changed.
+
+### Main Components of MyLedger
+`Parser:` Processes the inputs made by the user and converts into a sensible form for further processing.
+
+`Commands:` Matches the input command with the respective commands created and executes the command result.
+
+`Expenditure Type:` Expenditure information that allows the users to access their input data from the respective 
+command classes.
+
+`Storage:` Stores, reads and updates the user input into their hard disk.
+
+The following section describes the implementation of certain features.
+
+### 3.2. Parser
+#### Processing an input
+The main parser component `MainInputParser` is called whenever the user inputs a command line that requires action from 
+the application. The command word will be read and further processed into further components depending on the type of 
+command, such as `ParseAdd`, `ParseDelete`, `ParseIndividualValue` etc. 
+
+How the parsing works: 
+
+  `MainInputParser`:
+- The `MainInputParser` is called and is expected to return an object with the `Command` class.
+- The respective classes (eg `ParseAdd`, `ParseSort`) will be called.
+- For commands that requires an additional input after the command work (excludes commands like `list`, `help` etc), the
+  `ParseIndividualValue` class will be called to parse all the input fields by the user.
+
+The following shows the UML diagram used for the parser component implemented in MyLedger.
+
+<p align="center">
+    <img src="team/images/parserOverview.png">
```

Comment:
Image is cut off on the right side. Additionally, the figure is a little small due to the multiple options shown. One possible workaround could be just to include only 2 options just to show an example, while clearly stating that the other options are omitted from the diagram to prevent cluttering it.